### PR TITLE
Added an id generator that's more performant

### PR DESCRIPTION
**Pull request by bbakerman**
This is taken from Spring code to address #3435

https://github.com/spring-projects/spring-framework/blob/main/spring-core/src/main/java/org/springframework/util/AlternativeJdkIdGenerator.java

I am pretty sure under Apache 2.0 this is allowed

```
 4. Redistribution. You may reproduce and distribute copies of the
      Work or Derivative Works thereof in any medium, with or without
      modifications, and in Source or Object form, provided that You
      meet the following conditions:

      (a) You must give any other recipients of the Work or
          Derivative Works a copy of this License; and

      (b) You must cause any modified files to carry prominent notices
          stating that You changed the files; and

      (c) You must retain, in the Source form of any Derivative Works
          that You distribute, all copyright, patent, trademark, and
          attribution notices from the Source form of the Work,
          excluding those notices that do not pertain to any part of
          the Derivative Works; and

      (d) If the Work includes a "NOTICE" text file as part of its
          distribution, then any Derivative Works that You distribute must
          include a readable copy of the attribution notices contained
          within such NOTICE file, excluding those notices that do not
          pertain to any part of the Derivative Works, in at least one
          of the following places: within a NOTICE text file distributed
          as part of the Derivative Works; within the Source form or
          documentation, if provided along with the Derivative Works; or,
          within a display generated by the Derivative Works, if and
          wherever such third-party notices normally appear. The contents
          of the NOTICE file are for informational purposes only and
          do not modify the License. You may add Your own attribution
          notices within Derivative Works that You distribute, alongside
          or as an addendum to the NOTICE text from the Work, provided
          that such additional attribution notices cannot be construed
          as modifying the License.
```

4a - done
4b - done
4c - done
4d - not applicable

### CODE REVIEW

File: src/test/groovy/graphql/util/AlternativeJdkIdGeneratorTest.groovy [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+class AlternativeJdkIdGeneratorTest extends Specification {
+    def "can generate uuids"() {
+        when:
+        def uuid1 = AlternativeJdkIdGenerator.uuid()
+        def uuid2 = AlternativeJdkIdGenerator.uuid()
+        def uuid3 = AlternativeJdkIdGenerator.uuid()
+
+        then:
+        // should this fail - the universe has ended and has retracted back into the singularity
```
Comment (andimarek) [inline src/test/groovy/graphql/util/AlternativeJdkIdGeneratorTest.groovy:13]: make it bigger and make it a set and check the size

File: APACHE-LICENSE-2.0.txt [hunk 2]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (andimarek) [inline APACHE-LICENSE-2.0.txt:1]: I don't think this is good in the root folder ... it might confuse people about the license.

**Review by andimarek — CHANGES_REQUESTED** [review]

File: APACHE-LICENSE-2.0.txt (hunk 2 above)
Comment (bbakerman) [reply APACHE-LICENSE-2.0.txt]: done

File: src/test/groovy/graphql/util/AlternativeJdkIdGeneratorTest.groovy [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+class AlternativeJdkIdGeneratorTest extends Specification {
+    def "can generate uuids"() {
+        when:
+        def set = new HashSet()
+        for (int i = 0; i < 1000; i++) {
+            set.add(AlternativeJdkIdGenerator.uuid().toString());
+        }
+
+        then:
+        // should this fail - the universe has ended and has retracted back into the singularity
```
Comment (dondonz) [inline src/test/groovy/graphql/util/AlternativeJdkIdGeneratorTest.groovy:14]: Love it - we now have a sign of the apocalypse in this repo

**Comment by bbakerman** [comment]
>   [1 hour ago](https://graphql-java.slack.com/archives/C4V7R86KV/p1708925115820019?thread_ts=1708924250.098179&cid=C4V7R86KV)
looks good

**Pull request merged**